src/pages/CourseDetails.vue

```diff
@@ -0,0 +1,186 @@
+        </b-col>
+        <b-col
+          md="6"
+          sm="12"
+        >
+          <b-row>
+            <b-col>
+              <youtube
+                ref="youtube"
+                :video-id="course.introductoryVideoId"
+                width="100%"
+                height="400"
+              />
+            </b-col>
+          </b-row>
+          <b-row>
+            <b-col>
+              <div
+                ref="description"
+                class="course-description"
+                v-html="course.description"
+              />
+            </b-col>
+          </b-row>
+        </b-col>
+      </b-row>
+
+      <b-row v-if="!hideComments">
+        <b-col md="2" />
+        <b-col md="8">
+          <add-comment
+            ref="addcomment"
+            :comment-id="commentId"
+            :on-save="saveComment"
+            :on-cancel="cancelComment"
+            :show-rating="showRating"
+            :parent-id="course._id"
+            class="mt-1"
+          />
+          <Comment
+            v-for="(comment,index) in comments"
+            :key="comment._id"
+            :index="index"
+            :comment-id="comment._id"
+            :comment="comment"
+            :show-rating="showRating"
+            :allow-reply="allowReply"
+            :on-delete="deleteComment"
+            :on-edit="editComment"
+            :on-save="saveComment"
+          />
+        </b-col>
+        <b-col md="2" />
+      </b-row>
+    </b-container>
+  </div>
+</template>
+
+<script>
+import Comment from "@/components/Comment/Comment";
+import AddComment from "@/components/Comment/AddComment";
+import ChapterStrip from "@/components/Course/ChapterStrip";
+import commentService from "@/services/comment.service";
+import courseService from "@/services/course.service";
+import eventBus from "@/utilities/eventBus";
+import { ToastType, messages } from "@/constants/constants";
+export default {
+  name: "CourseDetails",
+  components: {
+    Comment,
+    AddComment,
+    ChapterStrip
+  },
+  props: {},
+  data() {
+    return {
+      course: {},
+      videoId: "7iUqMA2Y6xA",
+      hideComments: false,
+      showRating: false,
+      allowReply: true,
+      commentId: "",
+      comments: []
+    };
+  },
+  computed: {
+    signedInUser() {
+      return this.$store.state.signedInUser;
+    },
+    items() {
+      return [
+        {
+          text: "Courses",
+          to: "/learn/course"
+        },
+        {
+          text: this.course.title,
+          active: true
+        }
+      ];
+    }
+  },
+  mounted() {
+    this.loadCourse(this.$route.params.id);
+  },
+  created() {},
+  beforeDestroy() {
+    clearInterval(this.timer);
+  },
+
+  methods: {
+    loadCourse(courseId) {
+      courseService.getCoursesById(courseId).then(res => {
+        this.course = res;
+        this.getComments();
+      });
+    },
+    getComments() {
+      commentService
+        .getComment(this.course._id)
+        .then(response => {
```

Comment:
let's use async await at all places to be consistent. If it's too much work to change then leave it as it is for now